internal/transport/http2_server.go

```diff
@@ -53,10 +52,10 @@ import (
 var (
 	// ErrIllegalHeaderWrite indicates that setting header is illegal because of
 	// the stream's state.
-	ErrIllegalHeaderWrite = errors.New("transport: the stream is done or WriteHeader was already called")
+	ErrIllegalHeaderWrite = status.Error(codes.Internal, "transport: the stream is done or WriteHeader was already called")
```

Comment:
Please update this text since we're more precise about sending this error now.  E.g. `"stream.SendHeader called multiple times"`. (It's `SendHeader` at the user API level.)